src/providers/trusted_service/context/mod.rs

```diff
@@ -238,19 +241,15 @@ impl Context {
                 &mut resp_buf_size,
             )
         };
-        if status != 0 || opstatus != 0 {
+        Error::from_status_opstatus(status, opstatus).map_err(|e| {
```

Comment:
Do you need the `map_err` anymore?

Reply:
Yes, because I need to close the call if it's not successful

Reply:
oh yes missed that, I read `|e| e` as closure